### PR TITLE
[WP 5.9] Fix: HTML tags like inline images in nav links break submenu layout

### LINKED ISSUE #37656: FSE Navigation block: Inline images in nav links break submenu layout
### Description

Adding inline images to top level navigation items breaks the output of the submenu toggle indicator icon (see screen recording).

### Step-by-step reproduction instructions

1. Install a FSE Theme (e.g. TwentyTwentyTwo)
2. Edit site and add a new navigation
3. Add a submenu
4. Insert inline images to the top level navigation links
5. Save settings
6. Open the Frontend and notice the broken `.wp-block-navigation__submenu-icon` element.

### Screenshots, screen recording, code snippet

debug

Web Inspector:
inspector

### Environment info

- WordPress 5.8.2 (+ Gutenberg 12.2.0) OR WordPress 5.9 (RC 1)
- TwentyTwentyTwo Theme
- Chrome/Firefox (latest versions)

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by them-es**
Fixes https://github.com/WordPress/gutenberg/issues/37656
Once approved it needs to be ported to WordPress 5.9:
https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/blocks/navigation-submenu.php#L235-L239
https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/blocks/page-list.php#L186-L194

## Checklist:
- [x] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/block-library/src/navigation-submenu/index.php [hunk 1]
```diff
@@ -231,12 +231,12 @@ function render_block_core_navigation_submenu( $attributes, $content, $block ) {
 
 		if ( $show_submenu_indicators ) {
 			// The submenu icon is rendered in a button here
-			// so that there's a clickable elment to open the submenu.
-			$html .= '<button aria-label="' . $aria_label . '" class="wp-block-navigation__submenu-icon wp-block-navigation-submenu__toggle" aria-expanded="false">' . block_core_navigation_submenu_render_submenu_icon() . '</button>';
+			// so that there's a clickable element to open the submenu.
+			$html .= '<button aria-label="' . wp_strip_all_tags( $aria_label ) . '" class="wp-block-navigation__submenu-icon wp-block-navigation-submenu__toggle" aria-expanded="false">' . block_core_navigation_submenu_render_submenu_icon() . '</button>';
```
Comment (tellthemachines) [inline packages/block-library/src/navigation-submenu/index.php]: I'm thinking `wp_strip_all_tags` can instead be used on `$label` [here](https://github.com/WordPress/gutenberg/blob/062f5486f9f99a85944a9acc8fc5ce11d6257a03/packages/block-library/src/navigation-submenu/index.php#L194), where we generate the `$aria-label` content. That way we only need to use it once. 

We can do the same in the Page List file by using it on `$title` [here](https://github.com/WordPress/gutenberg/blob/062f5486f9f99a85944a9acc8fc5ce11d6257a03/packages/block-library/src/page-list/index.php#L180).

**Review by tellthemachines — COMMENTED** [review]
Thanks for fixing this! Just a suggestion for improvement below. 

I can't reproduce the e2e failures locally, so they may be unrelated.

File: packages/block-library/src/navigation-submenu/index.php (hunk 1 above)
Comment (them-es) [reply packages/block-library/src/navigation-submenu/index.php]: Thanks for reviewing this PR. It has been updated based on your suggestion.

**Review by tellthemachines — APPROVED** [review]
Thanks for the quick update 👍

**Pull request merged**